### PR TITLE
[TCES-21] Create JobApplication Model

### PR DESCRIPTION
**Pull request by harshithl1777**
# Description

- Created JobApplication Model
- Wiki update needed due to schema changes

Issue # TCES-21

## Type of change

Please delete options that are not relevant.

- [x] This change requires a documentation update

# Checklist

Please delete options that are not applicable.

- [x] My code follows the style guidelines of this project (required)
- [x] I have performed a self-review of my code (required)
- [x] I have linted and formatted my code (required)
- [x] New and existing unit tests pass locally with my changes (required)
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by 04kash — APPROVED** [review]
Looks good to me @miekoyao can you have a look as well since I remember some changes to the model were discussed?

File: server/src/models/job_applications.model.js
Comment (04kash) [inline server/src/models/job_applications.model.js:1]: Could you add an optional field called status_other, which would be a string to store a custom value when the user selects "other" as their status in Canada?
Details added to the ticket as well: https://linear.app/tces/issue/TCES-21/create-a-job-applications-model

**Comment by 04kash** [comment]
Added the status_other field in the latest commit

File: server/src/models/job_applications.model.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    resume: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    status_in_canada: {
+      type: DataTypes.ENUM(
+        "Citizen",
+        "PR",
+        "conventional refugee",
+        "student visa",
```
Comment (miekoyao) [inline server/src/models/job_applications.model.js]: The first letters of each of these options should be capitalized, similar to how capitalization is done for application_status

**Pull request merged**